src/components/YourCompany/YourCompany.vue

```diff
@@ -241,17 +244,19 @@ export default class YourCompany extends Mixins(DateMixin, EntityFilterMixin, Le
     }
   }
 
-  /** Emits Have Changes event. */
-  @Emit('haveChanges')
-  private emitHaveChanges (): boolean {
-    return (
-      this.companyNameChanges ||
+  private setDataChanges (): void {
+    const haveChanges: boolean = this.companyNameChanges ||
       this.contactInfoChanges ||
       this.folioNumberChanges ||
       this.nameTranslationChanges ||
       this.officeAddressChanges
-    )
+    this.setDefineCompanyStepChanged(haveChanges)
+    this.emitHaveChanges(haveChanges)
   }
+
+  /** Emits Have Changes event. */
+  @Emit('haveChanges')
+  private emitHaveChanges (haveChanges: boolean): void {}
```

Comment:
Why did you split up the original method into 2 methods?

Reply:
To set the change flag in the defineCompanyStep. I dont think it is right to set a state with in an emit method even though functionality will work that way

Reply:
OK